app/models/story.rb

```diff
@@ -92,6 +92,15 @@ class Story < ApplicationRecord
     top.order(score: :desc)
   }
 
+  scope :categories, ->(user, categories) {
+    tagged_story_ids = Tagging.select(:story_id).where(tag_id: Tag.where(category: categories).select(:id))
```

Comment:
I don't think I thought about this code before, but this and `tagged` look really inefficient. They fetch the list of every story id with the given tagging. There's now 15,000 stories tagged 'programming', and more every day.

Do you mind refactoring these two to use `.join` so the database can be smart about this? If you don't want to take it on, I can file a new issue for the performance work, but I figured I should ask as you now know these queries and tests best.